products/compute/api.yaml

```diff
@@ -2563,6 +2563,120 @@ objects:
         output: true
         description: |
           URL to a Network that should handle matching packets.
+  - !ruby/object:Api::Resource
+    name: 'Router'
+    kind: 'compute#router'
+    base_url: projects/{{project}}/regions/{{region}}/routers
+    exports:
+      - !ruby/object:Api::Type::SelfLink
+        name: 'selfLink'
+    description: |
+      Represents a Router resource.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+          guides:
+            'Google Cloud Router': 'https://cloud.google.com/router/docs/'
+          api: 'https://cloud.google.com/compute/docs/reference/rest/v1/routers'
+<%= indent(compile_file({}, 'templates/regional_async.yaml.erb'), 4) %>
+    parameters:
+      - !ruby/object:Api::Type::ResourceRef
+        name: region
+        resource: Region
+        imports: name
+        description: Region where the router resides.
+        input: true
+        required: true
+    properties:
+      - !ruby/object:Api::Type::Integer
+        name: 'id'
+        description: 'The unique identifier for the resource.'
+        output: true
+      - !ruby/object:Api::Type::Time
+        name: 'creationTimestamp'
+        description: 'Creation timestamp in RFC3339 text format.'
+        output: true
+      - !ruby/object:Api::Type::String
+        name: name
+        description: |
+          Name of the resource. The name must be 1-63 characters long, and
+          comply with RFC1035. Specifically, the name must be 1-63 characters
+          long and match the regular expression `[a-z]([-a-z0-9]*[a-z0-9])?`
+          which means the first character must be a lowercase letter, and all
+          following characters must be a dash, lowercase letter, or digit,
+          except the last character, which cannot be a dash.
+        input: true
+        required: true
+      - !ruby/object:Api::Type::String
+        name: description
+        description: |
+          An optional description of this resource.
+      - !ruby/object:Api::Type::ResourceRef
+        name: network
+        resource: Network
+        imports: 'selfLink'
+        description: |
+          A reference to the network to which this router belongs.
+        required: true
+        input: true
+      # TODO(danawillow): Figure out the story for interfaces/bgpPeers. Right
```

Comment:
That's a great model and I'd love to have it in these other providers.  Another of the places that MagicModules can be a better cleaner declarative version of the GCP APIs.